pkg/policies/policies.go

```diff
@@ -122,68 +127,84 @@ func (pv *PolicyVerifier) VerifyMaterial(ctx context.Context, material *v12.Atte
 	return result, nil
 }
 
-type PolicyEvaluation struct {
-}
-
 // VerifyStatement verifies that the statement is compliant with the policies present in the schema
 func (pv *PolicyVerifier) VerifyStatement(ctx context.Context, statement *intoto.Statement) ([]*v12.PolicyEvaluation, error) {
 	result := make([]*v12.PolicyEvaluation, 0)
 	policies := pv.schema.GetPolicies().GetAttestation()
 	for _, policyAtt := range policies {
 		// 1. load the policy spec
-		spec, ref, err := pv.loadPolicySpec(ctx, policyAtt)
+		policy, ref, err := pv.loadPolicySpec(ctx, policyAtt)
 		if err != nil {
 			return nil, NewPolicyError(err)
 		}
 
-		// it's expected statements can only be validated by policy of type ATTESTATION
-		if spec.GetSpec().GetType() != v1.CraftingSchema_Material_ATTESTATION {
+		// it's expected statements can only be validated by policies of type ATTESTATION
+		types := getPolicyTypes(policy)
+		if !slices.Contains(types, v1.CraftingSchema_Material_ATTESTATION) {
 			continue
 		}
 
-		// 2. load the policy script (rego)
-		script, err := LoadPolicyScriptFromSpec(spec)
+		// 2. load the policy scripts (rego)
+		scripts, err := LoadPolicyScriptsFromSpec(policy, v1.CraftingSchema_Material_ATTESTATION)
 		if err != nil {
 			return nil, NewPolicyError(err)
 		}
 
-		pv.logger.Info().Msgf("evaluating policy '%s' on attestation", spec.Metadata.Name)
-
 		material, err := protojson.Marshal(statement)
 		if err != nil {
 			return nil, NewPolicyError(err)
 		}
 
-		// 4. verify the policy
-		ng := getPolicyEngine(spec)
-		res, err := ng.Verify(ctx, script, material, getInputArguments(policyAtt.GetWith()))
+		pv.logger.Info().Msgf("evaluating policy '%s' on attestation", policy.Metadata.Name)
+
+		violations, scriptBody, err := pv.executeScript(ctx, policy, scripts, material, policyAtt)
```

Comment:
this method signagture is strange `pv.executeScript` in singular accepts multiple scripts. Why did you decide to do this instead of looping on the method and collecting the results here?

Reply:
Initially I thought of that, yes, that's why I accidentally kept the singular version of it. I'll extract the loop outside, I agree it would be easier to understand.

Reply:
Finally I kept the loop inside the method, just to reuse the most possible. Of course I changed the method name.